### PR TITLE
Uniq email #117

### LINKED ISSUE #117: User Story 20, User Editing Profile Data must have unique Email address
```
As a registered user
When I attempt to edit my profile data
If I try to change my email address to one that belongs to another user
When I submit the form
Then I am returned to the profile edit page
And I see a flash message telling me that email address is already in use
```

### Checklist:

- [ ] base code tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #121

depends on #120

depends on #118

### PR DESCRIPTION
**Pull request by stoic-plus**
* closes #117 

When updating user info with email that is in use
* redirect to `profile_edit_path`
* flash message indicating this

Adds `email_in_use?` to `User`

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -183,6 +183,13 @@
         expect(m.enabled).to eq(true)
       end
 
+      it '.email_in_use?' do
+        user1 = create(:user, email: "j@gmail.com")
+
+        expect(User.email_in_use?(user1.email)).to be_truthy
+        expect(User.email_in_use?("latin_lover@hotmail.com")).to be_falsy
```
Comment (le3ah) [inline spec/models/user_spec.rb]: 💃

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -31,20 +31,34 @@ def edit
...
+
   def user_params
     params.require(:user).permit(:name, :password, :email, :address, :city, :state, :zip)
   end
 
+  def update_user(user, params)
+    params.each do |attribute, value|
+      next if value.blank? || user[attribute] == value
+
+      if attribute =~ /password/
```
Comment (nicovigil1) [inline app/controllers/users_controller.rb]: =~?

Comment (nicovigil1) [reply app/controllers/users_controller.rb]: /password/?

File: app/models/user.rb [hunk 3]
```diff
@@ -52,6 +52,11 @@ def switch_enabled
       save
     end
 
+    def self.email_in_use?(email)
+      found_user = self.where(email: email)
+      !found_user.empty?
```
Comment (tonyr729) [inline app/models/user.rb]: `found_user.any?`

**Pull request merged**